File: programmers/lv0/코딩기초트레이닝/조건문/코드처리하기/Solution.go
package main

import (
	"fmt"
)

func solution(code string) string {
	answer := ""
	mode := 0
	for i := 0; i < len(code); i++ {
		if code[i] == '1' {
			mode = (mode + 1) % 2
		} else {
			if i%2 == mode {
				answer += string(code[i])
			}
		}
	}
	if len(answer) == 0 {
		return "EMPTY"
	}
	return answer
}

func main() {
	fmt.Println(solution("abc1abc1abc"))
}

File: programmers/lv0/코딩기초트레이닝/조건문/코드처리하기/Solution.cs
using System;

public class Solution {
    public string solution(string code) {
        string answer = "";
        int mode = 0;
        for (int i = 0; i < code.Length; i++)
        {
            char c = code[i];
            if (c == '1')
            {
                mode = mode == 0 ? 1 : 0;
                continue;
            }

            if (i % 2 == mode) answer += c;
        }
        return answer.Length == 0 ? "EMPTY": answer;
    }
}